shared/editor/nodes/Mention.ts

```diff
@@ -0,0 +1,206 @@
+import Token from "markdown-it/lib/token";
+import { InputRule } from "prosemirror-inputrules";
+import { NodeSpec, Node as ProsemirrorNode, NodeType } from "prosemirror-model";
+import { EditorState, TextSelection, Plugin } from "prosemirror-state";
+import { MarkdownSerializerState } from "../lib/markdown/serializer";
+import { run } from "../plugins/BlockMenuTrigger";
+import isInCode from "../queries/isInCode";
+import mentionRule from "../rules/mention";
+import { Dispatch, EventType } from "../types";
+import Node from "./Node";
+
+const OPEN_REGEX = /(?:^|\s)@([a-zA-Z+-]+)?$/;
+const CLOSE_REGEX = /(?:^|\s)@(([0-9a-zA-Z_+-]*\s+)|(\s+[0-9a-zA-Z_+-]+)|[^0-9a-zA-Z_+-]+)$/;
+
+export default class Mention extends Node {
+  get name() {
+    return "mention";
+  }
+
+  get schema(): NodeSpec {
+    return {
+      attrs: {
+        style: {
+          default: "",
+        },
+        "data-type": {
+          default: undefined,
+        },
+        "data-id": {
+          default: undefined,
+        },
+        "data-actor": {
+          default: undefined,
+        },
+        label: {
+          default: undefined,
+        },
+        id: {
+          default: undefined,
+        },
```

Comment:
```suggestion
        type: {},
        label: {},
        id: {
          default: undefined,
        },
        actorId: {
          default: undefined,
        },
```

"type" and "label" are required, the others can be optional

Reply:
> Lets add the mentionId= on here, even if it isn't supported today it will be

Let's include `mentionId` too in `attrs`?

Reply:
`id` is mentionId?

Reply:
`id` = identity of mentioned entity, like `userId`
`mentionId` = identify of the `mention` node itself which goes into the `id` HTML attribute of the resulting `span`.

Reply:
Ah, this could be quite easily confusing – perhaps `id` should be renamed to `modelId` then?

Reply:
:100: On it.

Reply:
And use `id` in place of `mentionId`?

Reply:
right